Tests/FunctionalTests/helpermethod.ts

```diff
@@ -147,3 +156,62 @@ export async function AcceptCookies(page: Page) {
     }
   }
 }
+
+export async function ExpectAllResultsHaveBatchUserAttValue(
+  page: Page, preciseValue: string): Promise<number> {
+
+  return await ExpectSelectionsAreEqual(page,
+    `//table[@class='${pageObjectsConfig.searchAttributeTable.substring(1)}']`,
+    `//table[@class='${pageObjectsConfig.searchAttributeTable.substring(1)}' and 0 < count(.//td[translate(text(),'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz')='${preciseValue.toLowerCase()}'])]`);
+}
+
+export async function ExpectAllResultsContainBatchUserAttValue(
+  page: Page, containsValue: string): Promise<number> {
+
+  return await ExpectSelectionsAreEqual(page,
+    `//table[@class='${pageObjectsConfig.searchAttributeTable.substring(1)}']`,
+    `//table[@class='${pageObjectsConfig.searchAttributeTable.substring(1)}' and 0 < count(.//td[contains(translate(text(),'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz'), '${containsValue.toLowerCase()}')])]`);
+}
+
+export async function ExpectAllResultsContainAnyBatchUserAttValue(
+  page: Page, containsOneOf: string[]): Promise<number> {
+
+  expect(containsOneOf.length).toBeTruthy();
+
+  const tdPredicate = containsOneOf
+    .map(containsValue => `contains(translate(text(),'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz'), '${containsValue.toLowerCase()}')`)
+    .join(' or ');
+
+  return await ExpectSelectionsAreEqual(page,
+    `//table[@class='${pageObjectsConfig.searchAttributeTable.substring(1)}']`,
+    `//table[@class='${pageObjectsConfig.searchAttributeTable.substring(1)}' and 0 < count(.//td[${tdPredicate}])]`);
+}
+
+export async function ExpectAllResultsHaveFileAttributeValue(
+  page: Page, preciseValue: string): Promise<number> {
+
+    return await ExpectSelectionsAreEqual(page,
+      `//table[@class='${pageObjectsConfig.fileAttributeTable.substring(1)}']`,
+      `//table[@class='${pageObjectsConfig.fileAttributeTable.substring(1)}' and 0 < count(.//td[translate(text(),'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz')='${preciseValue.toLowerCase()}'])]`);
+}
+
+async function ExpectSelectionsAreEqual(page: Page, tablePath: string, tablePathWithCondition: string): Promise<number> {
+  //  count the result rows
+  const resultCount = await page.$$eval(tablePath, matches => matches.length);
```

Comment:
Please can we amend this code to not use $$eval

Reply:
One for another time. Sorry